db/migrate/20250814153656_add_compound_primary_key_to_mhv_metrics_unique_user_events.rb

```diff
@@ -0,0 +1,14 @@
+class AddCompoundPrimaryKeyToMHVMetricsUniqueUserEvents < ActiveRecord::Migration[7.2]
+  disable_ddl_transaction!
+
+  def change
+    # Add unique compound index - columns already have NOT NULL constraints from table creation
+    # This ensures one record per user per event type
+    add_index :mhv_metrics_unique_user_events, 
+              [:user_id, :event_name], 
+              unique: true,
+              algorithm: :concurrently,
+              if_not_exists: true,
```

Comment:
I believe you can leave out these two lines since the table is created in the prior migration and does not yet contain any data.

Reply:
I see what you mean. I'll remove those. But, it won't hurt either.